File: cpp_solutions/2206.cpp
class Solution {
public:
    bool divideArray(vector<int>& nums) {
        vector<int> hash(501, 0);
        for (auto& num : nums) hash[num]++;
        for (int i = 0; i <= 500; ++i) {
            if (hash[i] & 1) return false;
        }
        return true;
    }
};

File: java_solutions/2206.java
class Solution {
    public boolean divideArray(int[] nums) {
        int[] hash = new int[501];
        for (int num : nums) hash[num]++;
        for (int i = 0; i <= 500; ++i) {
            if ((hash[i] & 1) == 1) return false;
        }
        return true;
    }
}
